src/lib.rs

```diff
@@ -147,34 +147,39 @@ pub use crate::primitives::absolute_locktime::{AbsLockTime, AbsLockTimeError};
 pub use crate::primitives::relative_locktime::{RelLockTime, RelLockTimeError};
 pub use crate::primitives::threshold::{Threshold, ThresholdError};
 
-/// Public key trait which can be converted to Hash type
+/// Trait representing a key which can be converted to a hash type.
 pub trait MiniscriptKey: Clone + Eq + Ord + fmt::Debug + fmt::Display + hash::Hash {
-    /// Returns true if the pubkey is uncompressed. Defaults to `false`.
+    /// Returns true if the key is serialized uncompressed, defaults to `false`.
     fn is_uncompressed(&self) -> bool { false }
 
-    /// Returns true if the pubkey is an x-only pubkey. Defaults to `false`.
+    /// Returns true if the key is an x-only pubkey, defaults to `false`.
     // This is required to know what in DescriptorPublicKey to know whether the inner
     // key in allowed in descriptor context
     fn is_x_only_key(&self) -> bool { false }
 
-    /// Returns the number of different derivation paths in this key. Only >1 for keys
-    /// in BIP389 multipath descriptors.
+    /// Returns the number of different derivation paths in this key, defaults to `0`.
+    ///
+    /// Only >1 for keys in BIP389 multipath descriptors.
     fn num_der_paths(&self) -> usize { 0 }
 
-    /// The associated [`bitcoin::hashes::sha256::Hash`] for this [`MiniscriptKey`], used in the
-    /// sha256 fragment.
+    /// The associated [`sha256::Hash`] for this `MiniscriptKey`, used in the sha256 fragment.
```

Comment:
In 12dd99d20223270524000ea9cd421fe7af7b132d:

We should just drop these doclinks. They don't make any sense.

Reply:
hmm, because I shorten the type it looses the `bitcoin::hashes` info which was in the link, you happy with that? It seems reasonable to link to `hashes`, what makes you say it doesn't make sense?

Reply:
Que? Do you mean that the point of mentioning it is just that its the sha256 algo and has nothing to do with the `hashes` crate? Or that we should not mention it at all?

Reply:
You can keep the reference to sha256, as in "The SHA-256 hash type used in the `sha256` fragment".

We can also drop the "this `MiniscriptKey`" which is redundant (it's alreday a method on the `MiniscriptKey` trait).